### ISSUE TITLE
Error running on `jest@29.4.0`

### ISSUE THREAD

**Issue by colinrotherham** [smooth-code/jest-puppeteer#512]
## 🐛 Bug Report

Hello 👋 

Running tests under `jest@29.4.0` throws the following error for

* **jest-puppeteer**
* **jest-environment-puppeteer**

```console
  ● Test suite failed to run

    TypeError: Cannot read properties of undefined (reading 'bind')

      at Runtime._createJestObjectFor (node_modules/jest-runtime/build/index.js:2099:64)
```

## To Reproduce

1. Upgrade to `jest@29.4.0`
2. Run tests

## Expected behavior

Tests to run as expected

## The fix

It appears that the older dependency [`jest-environment-node@27.5.1`](https://github.com/smooth-code/jest-puppeteer/pull/508/commits/f4830ddaa21feff143abc0ad4a50ab798c49e2a9#diff-2f06640d1e61c571f77b31657fff4c3aa14743d0ed3912a733ce8b7997c17311R27) is no longer compatible

But overriding locally via `package.json` works:

```json
"overrides": {
  "jest-environment-node": "^29.4.0"
}
```

Hope this helps

**Comment by colinrotherham** [comment]
Due to this project supporting Node.js 12+ (npm v6) it'd take a breaking change to move `jest-environment-node` to ~`dependencies`~ → `peerDependencies` where npm v7 would install it automatically.

Similarly raising the minimum version of Jest to **v27.0.4** would [include `jest-environment-node`](https://github.com/facebook/jest/commit/d6fb0d8fb0d43a17f90c7a5a6590257df2f2f6f5#diff-84dafc274ed45edb926c9b5e9c087fe5a96b167e006ad4bfa4e2421238075569):

```json
"jest-environment-jsdom": "^27.0.3",
"jest-environment-node": "^27.0.3",
```

That said, it looks like this may be already fixed in:

* https://github.com/facebook/jest/issues/13822

Released in `jest@29.4.1` 😊

**Comment by colinrotherham** [comment]
Confirming as fixed
